src/cli/llm/wizard.rs

```diff
@@ -40,13 +42,23 @@ impl<Q, A> Wizard<Q, A> {
 
     pub async fn ask(&self, q: Q) -> Result<A>
     where
-        Q: TryInto<ChatRequest, Error = super::Error>,
-        A: TryFrom<ChatResponse, Error = super::Error>,
+        Q: TryInto<ChatRequest, Error = Error> + Clone,
+        A: TryFrom<ChatResponse, Error = Error>,
     {
-        let response = self
-            .client
-            .exec_chat(self.model.as_str(), q.try_into()?, None)
-            .await?;
-        A::try_from(response)
+        let retry_strategy = ExponentialBackoff::from_millis(1000).map(jitter).take(5);
```

Comment:
the base for the strategy is too high, please, check the doc, it'll be `1 sec -> 16 minutes -> 277 hours -> ...` in that case.

Consider:
- change the base the way `base ^ attempt` is reasonable
- you may use the `factor` option to simplify the math
- specify `max_delay` to limit maximum delay
- don't use jitter, it adds randomness making it harder to mitigate 429 status